File: x-runtime/clojure/src/hermes/agentic.clj
(ns hermes.agentic
  "Agentic task queue/runtime."
  (:require [hermes.protocols :as p]
            [clojure.core.async :as async]))

(defn task [owner kind input]
  {:id (format "task-%s" (java.util.UUID/randomUUID))
   :owner owner
   :kind kind
   :input input
   :output nil
   :status :queued
   :created-at (p/now-ms)
   :updated-at (p/now-ms)})

File: x-runtime/typescript/src/agentic.ts
import { WorkerId } from './protocols';

export interface AgenticTask {
  id: string;
  owner: WorkerId;
  kind: string;
  input: unknown;
  output?: unknown;
  status: TaskStatus;
  createdAt: number;
  updatedAt: number;
}

export enum TaskStatus {
  Queued = 'queued',
  Running = 'running',
  Completed = 'completed',
  Failed = 'failed',
}

export function createTask(owner: WorkerId, kind: string, input: unknown): AgenticTask {
  const now = Date.now();
  return {
    id: `task-${crypto.randomUUID()}`,
    owner,
    kind,
    input,
    status: TaskStatus.Queued,
    createdAt: now,
    updatedAt: now,
  };
}

export class TaskRegistry {
  private readonly tasks = new Map<string, AgenticTask>();

  put(task: AgenticTask): void {
    this.tasks.set(task.id, task);
  }

  get(id: string): AgenticTask | undefined {
    return this.tasks.get(id);
  }
}
